src/Language/Fortran/Parser/Fixed/Fortran77.y

```diff
@@ -236,38 +236,36 @@ MAYBE_ID :: { Maybe Name }
 NAME :: { Name } : id { let (TId _ name) = $1 in name }
 
 INCLUDES :: { [ Block A0 ] }
-: maybe(NEWLINE) list(BLOCK) { $2 }
+: BLOCKS maybe(NEWLINE) { $1 }
 
 BLOCKS :: { [ Block A0 ] }
-: BLOCKS BLOCK { $2 : $1 }
+: BLOCKS NEWLINE BLOCK { $3 : $1 }
+| BLOCK { [ $1 ] }
 | {- EMPTY -} { [ ] }
 
 BLOCK :: { Block A0 }
-: IF_BLOCK NEWLINE { $1 }
-| LABEL_IN_6COLUMN STATEMENT NEWLINE { BlStatement () (getTransSpan $1 $2) (Just $1) $2 }
-| STATEMENT NEWLINE { BlStatement () (getSpan $1) Nothing $1 }
-| COMMENT_BLOCK { $1 }
+: IF_BLOCK { $1 }
+| LABEL_IN_6COLUMN STATEMENT { BlStatement () (getTransSpan $1 $2) (Just $1) $2 }
+| STATEMENT { BlStatement () (getSpan $1) Nothing $1 }
+| comment { let (TComment s c) = $1 in BlComment () s (Comment c) }
 
 IF_BLOCK :: { Block A0 }
-:                  if '(' EXPRESSION ')' then NEWLINE BLOCKS ELSE_BLOCKS
+: if '(' EXPRESSION ')' then BLOCKS NEWLINE ELSE_BLOCKS
```

Comment:
I think I'm misunderstanding something, but when I read this (and the `BLOCKS`, `BLOCK` rules), I don't see where the initial `NEWLINE` between `then` and the internal blocks gets consumed (which it must do, because all the tests are happy!).

Reply:
`BLOCKS` can be empty, so `BLOCKS NEWLINE BLOCK` can be `{- empty -}, newline, block`, hence everything being happy.

Reply:
Oh! It's caught by `BLOCKS NEWLINE BLOCK`, where the first `BLOCKS` is the empty case. No S/R or R/R conflicts, so seems good. I wonder what this might mean for performance though.